tests/integration/ha/test_horizontal_scaling.py

```diff
@@ -257,3 +274,60 @@ async def test_safe_scale_down_roles_reassigning(
 
     # continuous writes checks
     await assert_continuous_writes_consistency(c_writes)
+
+
+async def test_safe_scale_down_remove_leaders(
+    ops_test: OpsTest, c_writes: ContinuousWrites, c_writes_runner
+) -> None:
+    """Tests the removal of specific units (elected cluster_manager, juju leader).
+
+    The goal of this test is to make sure that:
+     - the CM reelection happens successfully.
+     - the leader-elected event gets triggered successfully and
+        leadership related events on the charm work correctly, i.e: roles reassigning
+    """
+    app = await app_name(ops_test)
+
+    # scale up by 2 units
+    await ops_test.model.applications[app].add_unit(count=2)
+    await ops_test.model.wait_for_idle(
+        apps=[app], status="active", timeout=1000, wait_for_exact_units=5, idle_period=60
+    )
+
+    # scale down: remove the juju leader
+    leader_unit_id = await get_leader_unit_id(ops_test)
+
+    await ops_test.model.applications[app].destroy_unit(f"{app}/{leader_unit_id}")
+    await ops_test.model.wait_for_idle(
+        apps=[app], status="active", timeout=1000, wait_for_exact_units=4, idle_period=120
```

Comment:
question: what is the reasoning behind these number specified - specifically for idle period

Reply:
On a scale-up/down event, there might occur a recomputing of the nodes' roles - then the restart of the impacted units. 
When this occurs, depending on the responsiveness of the charm hooks - it could be triggered up to a certain amount of time after being active. 
so to be safe `idle_period` is set, otherwise we could lose a unit (restarting) a few seconds after `wait_for_idle` returns true, making the rest of the test fail.